src/sentry/models/release.py

```diff
@@ -381,61 +381,70 @@ def set_commits(self, commit_list):
-                        authors[author_email] = author = CommitAuthor.objects.get_or_create(
+                        author_data = {
+                            'name': data.get('author_name')
+                        }
+                        author, created = CommitAuthor.objects.create_or_update(
                             organization_id=self.organization_id,
                             email=author_email,
-                            defaults={
-                                'name': data.get('author_name'),
-                            }
-                        )[0]
-                        if data.get('author_name') and author.name != data['author_name']:
-                            author.update(name=data['author_name'])
+                            defaults=author_data,
+                            values=author_data)
+                        if not created:
+                            author = CommitAuthor.objects.get(
+                                organization_id=self.organization_id,
+                                email=author_email)
+                        authors[author_email] = author
                     else:
                         author = authors[author_email]
 
-                    defaults = {
+                    commit_data = {
                         'message': data.get('message'),
-                        'author': author,
                         'date_added': data.get('timestamp') or timezone.now(),
                     }
-                    commit, created = Commit.objects.get_or_create(
+                    # If we didn't get an author don't overwrite an existing one.
+                    if author is not None:
+                        commit_data['author'] = author
+
+                    commit, created = Commit.objects.create_or_update(
                         organization_id=self.organization_id,
                         repository_id=repo.id,
                         key=data['id'],
-                        defaults=defaults,
-                    )
+                        defaults=commit_data,
+                        values=commit_data)
+                    if not created:
+                        commit = Commit.objects.get(
+                            organization_id=self.organization_id,
+                            repository_id=repo.id,
+                            key=data['id'])
+
                     if author is None:
                         author = commit.author
 
                     commit_author_by_commit[commit.id] = author
 
                     patch_set = data.get('patch_set', [])
-
                     for patched_file in patch_set:
-                        CommitFileChange.objects.get_or_create(
+                        commit_file_data = {
+                            'filename': patched_file['path'],
+                            'type': patched_file['type'],
+                        }
+                        CommitFileChange.objects.create_or_update(
```

Comment:
We never actually updated this before, might make sense to just catch the integrity error.